block-and-parry-demo/Game.gd

```diff
@@ -0,0 +1,55 @@
+extends Node2D
+
+var _freezing := false
+
+onready var _player := $Player
+onready var _enemy := $Enemy
+onready var _camera := $Camera2D
+onready var _tween := $Tween
+
+
+func _ready() -> void:
+	_player.connect("blocked", self, "_on_Player_blocked")
+	_player.connect("hit", self, "_on_Player_hit")
+	_enemy.connect("stunned", self, "_on_Enemy_stunned")
+
+
+func _shake_screen(strength: float, frequency: int, rate: float) -> void:
+	for i in frequency:
+		var rand_offset: Vector2 = Vector2.ZERO
+		rand_offset.x = rand_range(-strength, strength)
+		rand_offset.y = rand_range(-strength, strength)
+		
+		_tween.interpolate_property(_camera, "offset", _camera.offset, rand_offset, rate, Tween.TRANS_LINEAR, Tween.EASE_IN)
+		_tween.start()
+
+		yield(_tween, "tween_all_completed")
+	
+	_tween.interpolate_property(_camera, "offset", _camera.offset, Vector2.ZERO, rate, Tween.TRANS_LINEAR, Tween.EASE_IN)
+	_tween.start()
+	
+
+func _freeze_frame(duration: float, time_scale: float) -> void:
```

Comment:
This slows down time rather than really freezes the screen, so I would say `_slow_down_time` is a clearer name.